metricflow-semantics/metricflow_semantics/specs/spec_classes.py

```diff
@@ -672,6 +677,7 @@ class WhereFilterSpec(Mergeable, SerializableDataclass):
     where_sql: str
     bind_parameters: SqlBindParameters
     linkable_specs: Tuple[LinkableInstanceSpec, ...]
+    linkable_elements: Tuple[LinkableElement, ...]
```

Comment:
Not necessary for this PR, but I wonder if it would be cleaner to add some simple converters between linkable specs and linkable elements so that you don't have to pass both through everywhere.

Reply:
We've talked about consolidating these and I think we'll get there eventually. One of the issues we have right now is LinkableSpecs aren't convertible to LinkableElements because the latter contain a bunch of additional metadata that LinkableSpecs can't carry along with them, so there's no way to simply add that information without causing all kinds of havoc with existing spec comparisons (e.g., `some_entity_spec == some_other_entity_spec`).

For now I could pass the LinkableElementSet along but that gets slightly more complicated so I decided to wire the elements through. I might still switch over to passing the set to the end here. I'm honestly not sure which approach is less annoying to read.

Reply:
I could replace the linkable_specs with linkable_elements and make the former a property accessor that converts from the element to a spec, I suppose, but it's a bit gnarly without the LinkableInstanceSet.